File: infrastructures/ddd/csharp/src/Infrastructure/Repositories/IExampleQueryRepository.cs
using System.Collections.Generic;
using System.Threading.Tasks;
using OddlyDdd.Infrastructure.Persistence.Read;

namespace OddlyDdd.Infrastructure.Repositories
{
    /// <summary>
    /// Example query repository interface demonstrating CQRS query side.
    /// Query repositories:
    /// - Return ReadEntities directly to service
    /// - NO BMO mapping in read path (for performance)
    /// - Optimized for query patterns
    /// - Used for: Get, List, Search operations
    /// </summary>
    public interface IExampleQueryRepository : IQueryRepository<ExampleReadEntity, string>
    {
        /// <summary>
        /// Finds examples by owner ID.
        /// Example of a custom query method.
        /// </summary>
        Task<IReadOnlyList<ExampleReadEntity>> FindByOwnerIdAsync(string p_ownerId);

        /// <summary>
        /// Finds active examples only.
        /// Example of a custom query method.
        /// </summary>
        Task<IReadOnlyList<ExampleReadEntity>> FindActiveExamplesAsync(int p_skip, int p_take);
    }
}

File: infrastructures/ddd/java/src/main/java/com/oddly/ddd/infrastructure/repositories/IExampleQueryRepository.java
package com.oddly.ddd.infrastructure.repositories;

import com.oddly.ddd.infrastructure.persistence.read.ExampleReadEntity;
import com.oddly.ddd.infrastructure.repositories.infra.IQueryRepository;

import java.util.List;
import java.util.concurrent.CompletableFuture;

/**
 * Query repository interface for Example domain.
 * Query repositories:
 * - Live in /infrastructure/repositories/
 * - Have IQueryRepository suffix
 * - Implementations go in /infrastructure/repositories/impl/
 * - Work with ReadEntities
 * - Return ReadEntity directly to service (no BMO in read path)
 */
public interface IExampleQueryRepository extends IQueryRepository<ExampleReadEntity, String> {
    /**
     * Finds examples by category.
     * 
     * @param p_category The category to filter by
     * @return List of examples in the category
     */
    CompletableFuture<List<ExampleReadEntity>> findByCategoryAsync(String p_category);

    /**
     * Finds active examples.
     * 
     * @return List of active examples
     */
    CompletableFuture<List<ExampleReadEntity>> findActiveAsync();
}
